### PR TITLE
[Android] Rewrite remaining generation scripts

### PR DESCRIPTION
**Pull request by rakuco**
This should finish all the required rewrites to the `generate_*.py` scripts in `build/android`.

Not only do they make more sense now and look cleaner, but using them from GN is a lot easier.

This follows what was done in #3897: all the logic and decisions about file paths are now the responsibility of the build system, and the scripts just copy files around and put them in the right directories.

### CODE REVIEW

**Comment by rakuco** [comment]
@lincsoon @heke123

File: xwalk_core_library_android.gypi [hunk 1]
```diff
@@ -441,7 +441,13 @@
...
           'action': [
             'python', '<(DEPTH)/xwalk/build/android/generate_xwalk_core_library_aar.py',
-            '-t', '<(PRODUCT_DIR)',
+            '--aar-path', '<(PRODUCT_DIR)/xwalk_core_library.aar',
+            '--android-manifest', '<(DEPTH)/xwalk/build/android/xwalkcore_library_template/AndroidManifest.xml',
+            '--classes-jar', '<(PRODUCT_DIR)/lib.java/xwalk_core_library_java.jar',
+            '--jni-abi', '<(android_app_abi)',
+            '--jni-dir', '<(PRODUCT_DIR)/xwalk_core_library/libs/<(android_app_abi)',
+            '--res-dir', '<(PRODUCT_DIR)/xwalk_core_library/res',
+            '--r-txt', '<(PRODUCT_DIR)/xwalk_core_internal_empty_embedder_apk/gen/R.txt',
```
Comment (heke123) [inline xwalk_core_library_android.gypi]: In original logic it is "xwalk_core_empty_embedder_apk/gen/R.txt" but here it is "xwalk_core_**internal**_empty_embedder_apk"?

File: xwalk_core_library_android.gypi [hunk 2]
```diff
@@ -466,8 +472,11 @@
           ],
           'action': [
             'python', '<(DEPTH)/xwalk/build/android/generate_xwalk_core_library_aar.py',
-            '-t', '<(PRODUCT_DIR)',
-            '--shared',
+            '--aar-path', '<(PRODUCT_DIR)/xwalk_shared_library.aar',
+            '--android-manifest', '<(DEPTH)/xwalk/build/android/xwalk_shared_library_template/AndroidManifest.xml',
+            '--classes-jar', '<(PRODUCT_DIR)/lib.java/xwalk_core_library_java_app_part.jar',
+            '--res-dir', '<(PRODUCT_DIR)/xwalk_shared_library/res',
+            '--r-txt', '<(PRODUCT_DIR)/xwalk_core_internal_empty_embedder_apk/gen/R.txt',
```
Comment (heke123) [inline xwalk_core_library_android.gypi]: ditto

File: xwalk_core_library_android.gypi (hunk 1 above)
Comment (rakuco) [inline xwalk_core_library_android.gypi]: Fixed, thanks.

File: xwalk_core_library_android.gypi (hunk 2 above)
Comment (rakuco) [inline xwalk_core_library_android.gypi]: Also fixed.

Comment (heke123) [inline xwalk_core_library_android.gypi]: by the way, I didn't understand well on the meaning of                     
```
 'outputs': [
             '<(PRODUCT_DIR)/xwalk_core_library_aar_intermediate/always_run',
            ],
```
when to set it as "stamp" and when to "always_run"? thanks.

Comment (rakuco) [inline xwalk_core_library_android.gypi]: The "always_run" version is a hack that the previous people who worked on this file came up with -- you can see it's not present in Chromium itself. It basically points to a file that is never created, so the target is always run regardless of whether it needs to.
The stamp version is much more commonly used -- you pass a file that will be touched when the script finishes running, and the build system uses that to determine whether the inputs are more recent that it and only call the script if something changed.

Comment (heke123) [inline xwalk_core_library_android.gypi]: I know it has nothing to do with this PR.
Why can't we just remove the 'output' if no stamp is needed?  Is the 'output' is mandatory in "action" of GYP?

**Comment by rakuco** [comment]
Patch rebased on top of #3902.

File: xwalk_core_library_android.gypi (hunk 2 above)
Comment (rakuco) [inline xwalk_core_library_android.gypi]: `output` is mandatory for actions, just like in GN.

**Comment by heke123** [comment]
LGTM

**Pull request merged**